app/helpers/dynamic_form_helper.rb

```diff
@@ -61,6 +61,31 @@ def link_to_remove_nested_item(form)
     raw out
   end
 
+  def link_to_remove_support_file(form)
+    new_record = form.object.new_record?
```

Comment:
```suggestion
    persisted = form.object.persisted?
```
Es la inversa de `new_record?` y no usás `unless` abajo.